clippy_lints/src/transmute/missing_transmute_annotations.rs

```diff
@@ -0,0 +1,53 @@
+use rustc_errors::Applicability;
+use rustc_hir::{GenericArg, HirId, Node, Path, TyKind};
+use rustc_lint::LateContext;
+use rustc_middle::lint::in_external_macro;
+use rustc_middle::ty::Ty;
+
+use clippy_utils::diagnostics::span_lint_and_sugg;
+
+use crate::transmute::MISSING_TRANSMUTE_ANNOTATIONS;
+
+pub(super) fn check<'tcx>(
+    cx: &LateContext<'tcx>,
+    path: &Path<'tcx>,
+    from_ty: Ty<'tcx>,
+    to_ty: Ty<'tcx>,
+    expr_hir_id: HirId,
+) -> bool {
+    let last = path.segments.last().unwrap();
+    if in_external_macro(cx.tcx.sess, last.ident.span) {
```

Comment:
Makes me wonder if this shouldn't just apply to *all* transmute lints, much like how `methods/mod.rs` also blanket allows method calls from external macros. 🤔

Reply:
`transmute` includes lints that catch UB, so they should include external macros. Like `transmute_null_to_fn` as an example

Reply:
I can remove this limitation if you want. No opinion on this so whatever the clippy team prefers.

Reply:
Oh. So something outside of this lint. Might be worth opening an issue so it's not forgotten.